packages/react/src/returns/hooks/useUserReturns.ts

```diff
@@ -167,15 +175,38 @@ function useUserReturns(options: UseUserReturnsOptions = {}) {
       return Promise.reject(new Error('No userId provided'));
     }
 
+    if (useLegacyGuestFlow && !isGuest) {
+      return Promise.reject(
+        new Error('The user must be a guest to use the legacy flow'),
+      );
+    }
+
     const requestId = currentRequestId.current++;
     const actionMetadata = { userId, requestId, queryHash };
+    const getUserReturnsLegacyData =
```

Comment:
Shouldn't we validate if `useLegacyGuestFlow` is true but either `guestUserEmail` or `orderId` are undefined?

Reply:
Added the validation. Please take a look.